src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/exceptionhandlers/ErrorNotificationExceptionHandler.java

```diff
@@ -0,0 +1,48 @@
+package uk.gov.hmcts.reform.bulkscanprocessor.exceptionhandlers;
+
+import com.microsoft.azure.servicebus.IMessage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.gov.hmcts.reform.bulkscanprocessor.client.ErrorNotificationException;
+import uk.gov.hmcts.reform.bulkscanprocessor.model.ErrorNotificationMessageWrapper;
+
+import javax.validation.constraints.NotNull;
+
+public class ErrorNotificationExceptionHandler {
+
+    private static final Logger log = LoggerFactory.getLogger(ErrorNotificationExceptionHandler.class);
+
+    public ErrorNotificationExceptionHandler() {
+        // empty construct
+    }
+
+    public ErrorNotificationMessageWrapper handle(IMessage message, Throwable throwable) {
+        // return for message session completion step
+        return throwable != null
+            ? handleNonNullThrowable(message, throwable)
+            : ErrorNotificationMessageWrapper.forAcknowledgement(message);
+    }
+
+    private ErrorNotificationMessageWrapper handleNonNullThrowable(IMessage message, @NotNull Throwable throwable) {
+        try {
+            throw (RuntimeException) throwable;
```

Comment:
So it can be applied in the `handle` function in session handler. those are standard functional interfaces without checked exception

Reply:
can we just do something like below instead of casting it ?
```
 if (throwable instanceof RuntimeException) {
            return handleErrorNotificationException(message, (ErrorNotificationException) throwable);
        } else {
            log.error("Unable to cast Throwable to RuntimeException", throwable);
            return ErrorNotificationMessageWrapper.forDeadLettering(message);
        }
```